proto/ignition/msgs/wheel_slip_parameters_cmd.proto

```diff
@@ -31,9 +31,29 @@ message WheelSlipParametersCmd
   /// \brief Optional header data
   Header header                       = 1;
 
+  /// \brief Name of the model we want to modify the wheel slip parameters.
```

Comment:
I think that I'm not using scoped names in the current implementation (https://github.com/ignitionrobotics/ign-gazebo/pull/1241), should I be?
If it's common to reuse model names but have different parents for each model, maybe I should be using scoped names.

Reply:
Yup, it looks like you're treating them as unscoped names.

I think it would be good to support nested models in some way. I can imagine creating a single wheel model that's repeated multiple times inside the same vehicle, for example, and they all may end up with the same name. I can't think of a clean way to do it with just `model_name` and `link_name`. Maybe if `model_name` accepts scoped names, but the implementation will need to be updated. My suggestion would be to remove `string model_name` and `string link_name` in favor of `Entity link_entity`.

Reply:
> My suggestion would be to remove string model_name and string link_name in favor of Entity link_entity.

That sounds good to me.
Is [name](https://github.com/ignitionrobotics/ign-msgs/blob/b41812ffe70a09e166a22d579f3f9053bdf45e00/proto/ignition/msgs/entity.proto#L78) in that case scoped?

Reply:
Ha I thought that message was better suited to work with nested models but looking at it now I think it isn't. Just saw this [bit of code](https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo6/src/systems/user_commands/UserCommands.cc) used to get "top-level" (i.e. children of the world) entities using the message.

Now I can't think of a clean way of using the `Entity` message for nested models unless the `name` is scoped. So it ends up not being too different from what you have here, except that the user can alternatively pass the ID, or make the type explicit.

Reply:
> except that the user can alternatively pass the ID

Yes, this might be useful in some cases.

> or make the type explicit.

I think it's always going to be a link, so that's maybe not needed (?).

---

In the current approach, you can use the same link name in different models, and distinguish them using the model name.
Would that be possible if I start using `msgs::Entity`?
e.g. https://github.com/ignitionrobotics/ign-gazebo/blob/f5bb284884de9c5d4d7fa5614e53b528654b6611/examples/worlds/trisphere_cycle_wheel_slip.sdf#L283 and https://github.com/ignitionrobotics/ign-gazebo/blob/f5bb284884de9c5d4d7fa5614e53b528654b6611/examples/worlds/trisphere_cycle_wheel_slip.sdf#L685.

Ideally, I agree that supporting nested models would be the best.

Reply:
> I think it's always going to be a link, so that's maybe not needed (?).

I think it's mostly useful if there's a generic way to handle the `Entity` message which multiple systems can use. This plugin could handle an error in case the user specifies a joint, for example.

It would be helpful to distinguish between links and other entities, in case there are name overlaps. For example, it's common to have links and joints with the same name (i.e. `right_wheel`).

> Would that be possible if I start using msgs::Entity?

Only if the `name` is scoped, I believe. I think it would be reasonable to use the `Entity` message this way. By the way, the [entitiesFromScopedName](https://github.com/ignitionrobotics/ign-gazebo/blob/f5bb284884de9c5d4d7fa5614e53b528654b6611/include/ignition/gazebo/Util.hh#L55-L81) helper function should help.

Reply:
> Only if the name is scoped, I believe. I think it would be reasonable to use the Entity message this way. By the way, the [entitiesFromScopedName](https://github.com/ignitionrobotics/ign-gazebo/blob/f5bb284884de9c5d4d7fa5614e53b528654b6611/include/ignition/gazebo/Util.hh#L55-L81) helper function should help.

Sounds good, I can try that.

Reply:
I got that working, see https://github.com/ignitionrobotics/ign-msgs/pull/227/commits/7c15ad8da4cc5f164d7ae870187acea7f5a4a5d1 and https://github.com/ignitionrobotics/ign-gazebo/pull/1241/commits/2d98ab94a7000ca6173a930ccd2948823879c459.